design/adr/core/0002-Developer-Scripts-TaggingAndCleanup.md

```diff
@@ -0,0 +1,105 @@
+# Developers Script - Clearing and Tagging
+
+<!--ts-->
+
+- [Status](#status)
+- [Context](#context)
+- [Proposed Design](#proposed-design)
+- [Decision](#decision)
+- [Consequences](#consequences)
+- [References](#references)
+
+<!--te-->
+
+## Status
+
+Proposed
+
+## Context  
+
+`Developer Script` repository contains helper scripts used by the EdgeX developers in their everyday work, docker-compose files for all the releases, and scripts for gathering project metrics. <br />
+This repository have only one branch - `master`. 
+While docker-compose files reside in different directories named appropriately for every specific release, this is not the case with the scripts in the main directory.
+It is supposed that these scripts are ready to work with the latest version of the project, that could not be helpful for developers troubleshooting older versions. 
+Actually, significant part of the scripts are currently outdated and not applicable even with the latest version. <br />
+Here it comes the need of keeping track scripts files and theirs cleaning up and update.
+
+## Proposed Design 
+
+### Force using annotated tags
+
+By using `annotated tags` git history will be marked with the time versions have been released. Exp: git tag -a v2.0 -m "Releasing version v2.0" <br />
+Developers will have the knowledge which specific commit to use for specific EdgeX version. 
+This will make the troubleshooting possible for both - the older and the current versions of the project.
```

Comment:
Add that this tagging needs to be part of the release process. 

Also our release czar (@lranjbar ) should also be aware of this proposal. ;-)

Reply:
- We can gather the requirements to use the CI to automatically tag developer-scripts with a version. That will not be a problem.

- We don't currently version developer-scripts. I'm not convinced that this approach will not create more confusion. Which repository version will developer scripts match to? Edgex-go? We would need to make sure we follow the guidelines we have already laid out in the wiki: https://wiki.edgexfoundry.org/pages/viewpage.action?pageId=21823969

- Another note: Tags should not be made by developers. The LF has a requirement that git tags should be signed using the GPG key provided by the LF for our project. The CI environment has the ability to use the GPG key provided by the LF so the most recent tags are automatically signed. For older tags we had to open a ticket with the LF to fulfill this requirement.

Reply:
@lenny-intel ,@lranjbar thank you for your review :)

 I was thinking of ... whenever Core EdgeX services are released with Major or Minor versions (excluding Patches) (usually two times in the year), developers-scripts repository to be tagged with the appropriate version / codename  (edinburgh, fuji, geneva etc)  or a combination of both - the codename and the version number .

The developers-scripts tags should coincide with the Core EdgeX services release codenames/versions.  This will ease the developers to understand, when looking at developers-scripts `git history` when different significant Edgex versions have been released. 

My suggestion is this `tagging` procedure to start from now on, without adding older tags (or at most to add tag for `Fuji` release only)

It would be best if the tags are added automatically from the CI.  

>  Which repository version will developer scripts match to? Edgex-go?

Developers script repository should match edgex-go versions

Reply:
That makes sense. We just need to add the part about following the core services version. This tagging would only happen during a release.

Reply:
I will update the ADR in order to be clearer